### DISCUSSION TITLE
Is it possible to make a GidraDev Processor project with two languages?

### DISCUSSION THREAD

**Discussion by hippietrail** [NationalSecurityAgency/ghidra#7116] [Q&A] [answered]
Directory structure of Processors in Ghidra: 
<img width="230" alt="image" src="https://github.com/user-attachments/assets/f31a3081-9042-4c0c-9013-fbf6bb711af0">

Directory structure of a new Processor project in GhidraDev within the workspace/project directory: 
<img width="254" alt="image" src="https://github.com/user-attachments/assets/6e01bc67-d820-41f7-a4e6-3ce206f2d70d">

We can see in Ghidra itself the same directory structure is repeated for each processor, but in GidraDev the one processor is basically at the top level of the project?

I don't know much about Java and Eclipse and Gradle as to how much info is in settings files or build system files vs discovered in the build process from the directory layout. Can we just change the structure arbitrarily, in one specific way, or not at all, to have a single extension project with two processors?

----

Actually, I just noticed that the filenames are the same and only the file extensions differ. Which may or may not mean we can just put two processors right into the same directory, which would be different to how Ghidra itself does it. I haven't tried yet to see if it "just works".

----

If more than one approach works, would one be preferred?

**Comment by GhidorahRex** [comment, marked as answer]
You can put multiple processors in the same directory - even with different `.ldefs` files. We have several processors like that like the AVR and the PIC. Usually we group processors together if they're part of the same processor line or family, but there's some exceptions and we handle those on a case-by-case basis. If you have a specific processor that you're wondering about, feel free to ask.
